src/main/java/net/dries007/tfc/world/classic/worldgen/vein/VeinType.java

```diff
@@ -214,4 +227,61 @@ public IBlockState getOreState(Rock rock, Ore.Grade grade)
             return this.oreState;
         }
     }
+
+    public enum Rarity
```

Comment:
What is the point of this? Rarity can already be displayed as an integer, and you're enforcing names upon specific values that may have no applicability to whatever custom veins someone is using.

Reply:
Seems you lost the point when i was talking about it in discord. Plain number does nothing and is horrible to the average user, while names are actually helpful to them. Here are both screenshots when i was talking about it:
![screen1](https://user-images.githubusercontent.com/22617477/77234886-817ae900-6b90-11ea-8ae5-91adc9cf8558.png)
![screen2](https://user-images.githubusercontent.com/22617477/77234888-83dd4300-6b90-11ea-8474-00c3f03cd390.png)

And yes, I can just remove it and show nothing, but why not give more info to the player if possible?

Reply:
From discord:
> but what does "Rare" mean

This is exactly the point. You aren't showing more information - you're just reducing the resolution of the rarity. You could well display "Rarity: 3%", (taking `1 / rarity`) rather than a value, but "Rare", or any word you choose, is obscuring meaning and also, *completely dependent on a specific set of veins*.